test/e2e/e2e.go

```diff
@@ -617,6 +617,143 @@ var _ = Describe("validation", func() {
 			return errors.IsNotFound(err)
 		}, timeout, interval).Should(BeTrue())
 	})
+
+	Context("Testing Update AddressPool", func() {
+		By("Creating addresspool object ", func() {
+			addresspool := &metallbv1alpha1.AddressPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "addresspool1",
+					Namespace: OperatorNameSpace,
+				},
+				Spec: metallbv1alpha1.AddressPoolSpec{
+					Protocol: "layer2",
+					Addresses: []string{
+						"1.1.1.1",
+						"1.1.1.100",
+					},
+				},
+			}
+
+			Expect(testclient.Client.Create(context.Background(), addresspool)).Should(Succeed())
+
+			key := types.NamespacedName{
+				Name:      "addresspool1",
+				Namespace: OperatorNameSpace,
+			}
+			// Create addresspool resource
+			By("By checking AddressPool resource is created")
+			Eventually(func() error {
+				err := testclient.Client.Get(context.Background(), key, addresspool)
+				return err
+			}, timeout, interval).Should(Succeed())
+
+			// Checking ConfigMap is created
+			By("By checking ConfigMap is created and matches addresspool configuration")
+			Eventually(func() (string, error) {
+				configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+				if err != nil {
+					return "", err
+				}
+				return configmap.Data[consts.MetalLBConfigMapName], err
+			}, timeout, interval).Should(MatchYAML(`address-pools:
+- name: addresspool1
+  protocol: layer2
+  addresses:
+
+  - 1.1.1.1
+  - 1.1.1.100
+
+`))
+
+		})
+
+		By("Update the same addresspool object with different range ", func() {
+			addresspool := &metallbv1alpha1.AddressPool{}
+			key := types.NamespacedName{
+				Name:      "addresspool1",
+				Namespace: OperatorNameSpace,
+			}
+			Eventually(func() error {
```

Comment:
Why do we do this get here? Can't we just create the new instance?

Reply:
I am reading the existing resource to update it, if u create another one with the same name u will get an error saying it already exists